sotodlib/coords/sat_mapmake.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+import so3g
+from pixell import enmap
+
+from .. import core, tod_ops, coords
+
+def make_map(tod,
+            wcs_kernel = None,
+            res = 0.1*coords.DEG,
+            dsT = None, demodQ = None, demodU = None,
+            cuts = None,
+            det_weights_predemod = None,
+            det_weights_dsT = None, det_weights_demod = None):
+    """
+    Generates maps of temperature and polarization from a TOD.
+
+    Parameters
+    ----------
+    tod : dict
+        An AxisManager object
+    wcs_kernel : enlib.wcs.WCS or None, optional
+        The WCS object used to generate the output map.
+        If None, a new WCS object with a Cartesian projection and a resolution of `res` will be created.
+    res : float, optional
+        The resolution of the output map, in radian.
+    dsT : array-like or None, optional
+        The input dsT timestream data. If None, the 'dsT' field of `tod` will be used.
+    demodQ : array-like or None, optional
+        The input demodulated Q timestream data. If None, the 'demodQ' field of `tod` will be used.
+    demodU : array-like or None, optional
+        The input demodulated U timestream data. If None, the 'demodU' field of `tod` will be used.
+    cuts : RangesMatrix or None, optional
+        A RangesMatrix that identifies samples that should be excluded from projection operations.
+        If None, no cuts will be applied.
+    det_weights_dsT : array-like or None, optional
+        The detector weights to use in the map-making for the dsT timestream.
+    det_weights_demod : array-like or None, optional
+        The detector weights to use in the map-making for the demodulated Q and U timestreams.
+        If both of `det_weights_dsT` and `det_weights_demod` are None, uniform detector weights will be used.
+        If only one of two are provided, the other weight is provided by `det_weights_dsT` = 2 * `det_weights_demod`.
+
+    Returns
+    -------
+    mTQU : enmap.ndmap
+        map of temperature and polarization
+    mTQU_weighted : enmap.ndmap
+        The inverse variance weighted map of temperature and polarization
+    wTQU : enmap.ndmap
+        The map of inverse variance weights used in the map-making process.
+    """
+    if wcs_kernel is None:
+        wcs_kernel = coords.get_wcs_kernel('car', 0, 0, res)    
+    if dsT is None:
+        dsT = tod['dsT']
+    if demodQ is None:
+        demodQ = tod['demodQ']
+    if demodU is None:
+        demodU = tod['demodU']
+    
+    PQU = coords.P.for_tod(tod=tod, wcs_kernel=wcs_kernel, cuts=cuts, comps='QU')
```

Comment:
Can you make PQU an optional argument to this function (but call it `P`).  The reason is that we might want to map in a different coordinate system (such as source-centered or planet-scan).  If P is provided as an argument, you don't need to check wcs_kernel.